runtime/lua/vim/lsp/util.lua

```diff
@@ -683,6 +684,53 @@ do
       api.nvim_buf_set_virtual_text(bufnr, diagnostic_ns, line, virt_texts, {})
     end
   end
+  function M.buf_diagnostics_statusline(bufnr, diagnostics)
+    all_buffer_diagnostics_counts[bufnr] = { errors=0, warnings=0, hints=0, info=0 }
+    for _, diagnostic in ipairs(diagnostics) do
+      if diagnostic.severity == 1 then
+        all_buffer_diagnostics_counts[bufnr]['errors'] = all_buffer_diagnostics_counts[bufnr]['errors'] + 1
+      elseif diagnostic.severity == 2 then
+        all_buffer_diagnostics_counts[bufnr]['warnings'] = all_buffer_diagnostics_counts[bufnr]['warnings'] + 1
+      elseif diagnostic.severity == 3 then
+        all_buffer_diagnostics_counts[bufnr]['info'] = all_buffer_diagnostics_counts[bufnr]['info'] + 1
+      elseif diagnostic.severity == 4 then
+        all_buffer_diagnostics_counts[bufnr]['hints'] = all_buffer_diagnostics_counts[bufnr]['hints'] + 1
+      end
+    end
+    vim.api.nvim_command("doautocmd User LSPStatusLineChanged")
```

Comment:
so this needs to be documented, right? We are using `Lsp` spelling (instead of `LSP`) for e.g. `:LspInstall` command, so let's use that for this autocmd name too.

Reply:
Thanks, will change to `LspDiagnosticsChanged` and will add it to `lsp.txt`